hack/windows/Install-OVS.ps1

```diff
@@ -187,33 +127,272 @@ function InstallOVS() {
+    $existingVCRedists = getInstalledVcRedists
+    foreach ($redist in $existingVCRedists) {
+        $installedVersion = [version]$redist.Version
+        # VC redists files with a higher version are installed, return.
+        if ($installedVersion -ge $mininalVersion) {
+            return
+        }
     }
-    Log "Hyper-V Virtual Machine Management service status: $VMMSStatus"
-    if ($VMMSStatus -eq "Running") {
-        cmd /c "cd $OVSDriverDir && install.cmd"
-    } else {
-        cd $OVSDriverDir ; netcfg -l .\ovsext.inf -c s -i OVSExt; cd $WorkDir
+    if (-not (Test-Path $VCRedistPath)) {
+        mkdir -p $VCRedistPath
+        curl.exe -Lo $VCRedistPath\vc_redist.x64.exe $DefaultVCRedistsDownloadURL
     }
-    if (!$?) {
-        Log "Install OVS kernel driver failed, exit"
+    # Install the VC redistributable files.
+    Get-ChildItem $VCRedistPath -Filter *.exe | ForEach-Object {
+        Start-Process -FilePath $_.FullName -Args '/install /passive /norestart' -Verb RunAs -Wait
+    }
+}
+
+function CheckAndInstallOVSDriver {
+    param (
+        [Parameter(Mandatory = $true)]
+        [String]$OVSDriverPath
+    )
+
+    $expVersion = [version]$(Get-Item $OVSDriverPath\ovsext.sys).VersionInfo.ProductVersion
+    $ovsInstalled = $(netcfg -q ovsext) -like "*is installed*"
+    $installedDrivers = getInstalledOVSDrivers
+
+    # OVSext driver with the desired version is already installed, return
+    if ($ovsInstalled -and (@($installedDrivers).Length -eq 1) -and ($installedDrivers[0].DriverVersion -eq $expVersion)){
+        return
+    }
+
+    $global:updateOVS = $true
+    # Uninstall the existing driver which is with a different version.
+    if ($ovsInstalled) {
+        netcfg -u ovsext
+    }
+
+    # Clean up the installed ovsext drivers packages.
+    foreach ($driver in $installedDrivers) {
+        $publishdName = $driver.PublishedName
+        pnputil.exe -d $publishdName
+    }
+
+    # Import OVSext driver certificate to TrustedPublisher and Root.
+    $DriverFile="$OVSDriverPath\ovsext.sys"
+    $CertificateFile = "$OVSDriverPath\package.cer"
+    $ExportType = [System.Security.Cryptography.X509Certificates.X509ContentType]::Cert
+    $Cert = (Get-AuthenticodeSignature $DriverFile).SignerCertificate
+    [System.IO.File]::WriteAllBytes($CertificateFile, $Cert.Export($ExportType))
+    Import-Certificate -FilePath "$CertificateFile" -CertStoreLocation cert:\LocalMachine\TrustedPublisher
+    Import-Certificate -FilePath "$CertificateFile" -CertStoreLocation cert:\LocalMachine\Root
+
+    # Install the OVSext driver with the desired version
+    # Copy $OVSDriverPath to host path "c:/openvwitch/driver", and then install ovsext.inf with the host path.
+    # This is a workaround for error code "0x80070003" with containerd v1.7+. The error happens when Windows
+    #  utitilty netcfg tries to access the container's mount path "C:/hpc/".
+    $hostDriverPath="C:\openvswitch\driver"
```

Comment:
Here also it feels like it should not be hardcoded, when we have `OVSInstallDir` as a parameter?

Reply:
Here the hard-code path is as expected, because for containerize case, `$OVSDriverPath` is a relative path inside the container, and we need an absolute path on the host to copy the driver files, otherwise the OVSext driver is auto-removed after the container is down, then we need to install it again and again.